test/cmd/exec.sh

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+run_kubectl_exec_pod_tests() {
+  set -o nounset
+  set -o errexit
+
+  create_and_use_new_namespace
+  kube::log::status "Testing kubectl exec POD COMMAND"
+
+  ### Test execute non-existing POD
+  output_message=$(! kubectl exec abc date 2>&1)
+  # POD abc should error since it doesn't exist
+  kube::test::if_has_string "${output_message}" 'pods "abc" not found'
+
+  ### Test execute existing POD
+  # Create test-pod
+  kubectl create -f hack/testdata/pod.yaml
+  # Execute existing POD
+  output_message=$(! kubectl exec test-pod date 2>&1)
+  # POD test-pod is exists this is shouldn't have output not found
+  kube::test::if_has_not_string "${output_message}" 'pods "test-pod" not found'
+  # These must be pass the validate
+  kube::test::if_has_not_string "${output_message}" 'pod or type/name must be specified'
+
+  # Clean up
+  kubectl delete pods test-pod
+
+  set +o nounset
+  set +o errexit
+}
+
+run_kubectl_exec_resource_name_tests() {
+  set +o nounset
+  set +o errexit
```

Comment:
Shouldn't this be -o?

Reply:
Or you need to avoid exiting when you test failure scenarios?

Reply:
@kargakis I just following other test cmd